Yoast/wordpress-seo
# Remove old social previews

## Context


* We don't need anything from the old social previews other than a target div for the React portal to render in, and hidden fields to persist the data on post/term save. This PR removes old code, makes the old social input fields hidden, and (re)introduces the Facebook / Twitter disabled/enabled functionality to React.

## Summary


This PR can be summarized in the following changelog entry:

* The `class-social-admin` will be deprecated.
* The old Social Previews (which were generated in PHP) were replaced by Social Previews rendered with React.
* The input fields of the old Social Previews are now hidden, and still used to save data.

## Relevant technical choices:

* The React social tab was always rendered with both Facebook and Twitter collapsibles and containers enabled. There is however a setting to disable Facebook/Twitter independently. Because this setting removes the (new) hidden fields for the disabled social platform, but React still rendered the new previews, errors occurred.
* We have now exposed the social settings via `wpseoScriptData.metabox.showSocial.facebook` and `wpseoScriptData.metabox.showSocial.twitter` booleans.
* The React components will render conditionally depending on those booleans.

## Test instructions

This PR can be tested by following these steps:

* Make sure there are no errors (PHP / JS console) throughout the test
* Verify that you now only see the new components in post edit and term edit.
* Verify that values in the social fields are saved on post/term save, and retrieved when you load the page again.
* In the social settings, make sure that disabling either Facebook, Twitter, or both, indeed removes the collapsibles. Disabling both should remove the Social tab in its entirety.
* Also test for premium, by checking out https://github.com/Yoast/wordpress-seo-premium/pull/2806 in the premium repo, and using
```
git fetch --all
git merge free/14428-remove-old-social-previews --no-commit
```
(you may have to fix a small merge conflict in the metabox formatter)

## UI changes

## Documentation

## Quality assurance


Fixes #14428
